src/main/java/org/geysermc/discordbot/commands/administration/SettingsCommand.java

```diff
@@ -38,70 +46,146 @@
 import java.util.Arrays;
 import java.util.List;
 
-public class SettingsCommand extends Command {
+public class SettingsCommand extends SlashCommand {
 
     public SettingsCommand() {
         this.name = "settings";
         this.hidden = true;
+        this.help = "Customize the bot's settings";
+        this.guildOnly = false;
+
         this.userPermissions = new Permission[] { Permission.MESSAGE_MANAGE };
+        this.botPermissions = new Permission[] { Permission.MESSAGE_MANAGE };
+
+        List<Command.Choice> types = new ArrayList<>();
+        List<Command.Choice> settings = new ArrayList<>();
+
+        types.add(new Command.Choice("add","add"));
+        types.add(new Command.Choice("get", "get"));
+        types.add(new Command.Choice("set", "set"));
+        types.add(new Command.Choice("remove","remove"));
+
+        settings.add(new Command.Choice("Allowed Invites", "allowed-invites"));
+        settings.add(new Command.Choice("Banned domains","banned-domains"));
+        settings.add(new Command.Choice("Banned IPs", "banned-ips"));
+        settings.add(new Command.Choice("Check domains", "check-domains"));
+        settings.add(new Command.Choice("Convert Extensions", "convert-extensions"));
+        settings.add(new Command.Choice("Don't level", "dont-level"));
+        settings.add(new Command.Choice("Don't log","dont-log"));
+        settings.add(new Command.Choice("Health Checks", "health-checks"));
+        settings.add(new Command.Choice("Log channel", "log-channel"));
+        settings.add(new Command.Choice("Punishment Message","punishment-message"));
+        settings.add(new Command.Choice("Roles","roles"));
+        settings.add(new Command.Choice("RSS Feeds", "rss-feeds"));
+        settings.add(new Command.Choice("Update channel", "update-channel"));
+        settings.add(new Command.Choice("Voice Role", "voice-role"));
+
+
+        this.options = Arrays.asList(
+                new OptionData(OptionType.STRING, "action", "The action to perform").setRequired(true).addChoices(types),
+                new OptionData(OptionType.STRING, "key", "The setting name").setRequired(true).addChoices(settings),
+                new OptionData(OptionType.STRING, "value", "The value to set").setRequired(false)
+        );
+    }
+
+    @Override
+    protected void execute(SlashCommandEvent event) {
+        // Defer to wait for us to handle the command
+        InteractionHook interactionHook = event.deferReply().complete();
+
+        // Fetch values
+        String action = event.getOption("action").getAsString();
+        String key = event.getOption("key").getAsString();
+        String value;
```

Comment:
```suggestion
        String value = event.optString("value", null);
```

Will remove your if/else below.